standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -0,0 +1,10189 @@
+import static org.apache.hadoop.hive.metastore.Warehouse.DEFAULT_CATALOG_NAME;
+import static org.apache.hadoop.hive.metastore.Warehouse.DEFAULT_DATABASE_COMMENT;
+import static org.apache.hadoop.hive.metastore.Warehouse.DEFAULT_DATABASE_NAME;
+import static org.apache.hadoop.hive.metastore.Warehouse.getCatalogQualifiedTableName;
+import static org.apache.hadoop.hive.metastore.utils.MetaStoreUtils.CAT_NAME;
+import static org.apache.hadoop.hive.metastore.utils.MetaStoreUtils.DB_NAME;
+import static org.apache.hadoop.hive.metastore.utils.MetaStoreUtils.getDefaultCatalog;
+import static org.apache.hadoop.hive.metastore.utils.MetaStoreUtils.parseDbName;
+import static org.apache.hadoop.hive.metastore.utils.MetaStoreUtils.prependCatalogToDbName;
+import static org.apache.hadoop.hive.metastore.utils.MetaStoreUtils.prependNotNullCatToDbName;
+
+public class HMSHandler extends FacebookBase implements IHMSHandler {
+  public static final Logger LOG = LoggerFactory.getLogger(HMSHandler.class);
+  private final Configuration conf; // stores datastore (jpox) properties,
+                                   // right now they come from jpox.properties
+
+  // Flag to control that always threads are initialized only once
+  // instead of multiple times
+  private final static AtomicBoolean alwaysThreadsInitialized =
+      new AtomicBoolean(false);
+
+  private static String currentUrl;
+  private FileMetadataManager fileMetadataManager;
+  private PartitionExpressionProxy expressionProxy;
+  private StorageSchemaReader storageSchemaReader;
+  private IMetaStoreMetadataTransformer transformer;
+
+  // Variables for metrics
+  // Package visible so that HMSMetricsListener can see them.
+  static AtomicInteger databaseCount, tableCount, partCount;
+
+  public static final String PARTITION_NUMBER_EXCEED_LIMIT_MSG =
+      "Number of partitions scanned (=%d) on table '%s' exceeds limit (=%d). This is controlled on the metastore server by %s.";
+
+  // Used for testing to simulate method timeout.
+  @VisibleForTesting
+  static boolean TEST_TIMEOUT_ENABLED = false;
+  @VisibleForTesting
+  static long TEST_TIMEOUT_VALUE = -1;
+
+  public static final String ADMIN = "admin";
+  public static final String PUBLIC = "public";
+
+  static final String NO_FILTER_STRING = "";
+  static final int UNLIMITED_MAX_PARTITIONS = -1;
+
+  private Warehouse wh; // hdfs warehouse
+  private static final ThreadLocal<RawStore> threadLocalMS =
+      new ThreadLocal<RawStore>() {
+        @Override
+        protected RawStore initialValue() {
+          return null;
+        }
+      };
+
+  private static final ThreadLocal<TxnStore> threadLocalTxn = new ThreadLocal<TxnStore>() {
+    @Override
+    protected TxnStore initialValue() {
+      return null;
+    }
+  };
+
+  private static final ThreadLocal<Map<String, com.codahale.metrics.Timer.Context>> timerContexts =
+      new ThreadLocal<Map<String, com.codahale.metrics.Timer.Context>>() {
+    @Override
```

Comment:
Please fix the indentation. There are a lot of other indentation issues introduced since the last version (also some fixed). Please run a checkstyle check (mvn checkstyle:checkstyle) on the metastore-server project, and fix at least the indentation issues in the HMSHandler class (preferably all the other issues too).

Reply:
Fixed the indentation in HMSHandler and HiveMetaStore. Some of the issues in HMSHandler stem from mismatch of Thrift method names and checkstyle rules, I don't think they are touchable. I fixed many others, WDYT now?

Reply:
Sure, no need to fix those. Thanks!